packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -109,228 +111,225 @@ type AccountOf = {
 };
 
 type PositionInfo = EVMProtocolState | NobleDollarState;
+type ProtocolKey = keyof typeof YieldProtocol;
+
 type PortfolioKitState = {
-  positions: MapStore<number, PositionInfo>;
+  positions: MapStore<ProtocolKey, PositionInfo>;
+  gmp: MapStore<'manager', AxelarGmpManager>;
 };
 
-export const preparePortfolioKit = (zone: Zone, { zcf }: { zcf: ZCF }) =>
+export const preparePortfolioKit = (
+  zone: Zone,
+  {
+    chainHub,
+    timer,
+    zcf,
+    axelarChainsMap,
+    vowTools,
+  }: {
+    chainHub: ChainHub;
+    zcf: ZCF;
+    axelarChainsMap: AxelarChainsMap;
+    timer: Remote<TimerService>;
+    vowTools: VowTools;
+  },
+) =>
   zone.exoClassKit(
     'Portfolio',
     {
       keeper: KeeperI,
       holder: HolderI,
       invitationMakers: M.interface('invitationMakers', {
-        supplyToAave: M.call(M.record()).returns(M.promise()),
-        withdrawFromAave: M.call(M.record()).returns(M.promise()),
+        supplyToAave: M.call(M.record()).returns(VowShape),
+        withdrawFromAave: M.call(M.record()).returns(VowShape),
```

Comment:
a `Rebalance` invitationMaker seems necessary and sufficient.

Note that invitation maker method names conventionally start with a capital letter. I forget exactly why... something about not clashing with JS built-ins... IOU a link.